remme/rpc_api/_base.py

```diff
@@ -0,0 +1,238 @@
+# Copyright 2018 REMME
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------
+import logging
+import weakref
+import asyncio
+import threading
+from contextlib import contextmanager, suppress
+
+import aiohttp
+from aiohttp import web
+
+from aiohttp_json_rpc.rpc import JsonRpc
+from aiohttp_json_rpc.protocol import (
+    JsonRpcMsgTyp,
+    encode_result,
+    encode_error,
+    decode_msg,
+)
+from aiohttp_json_rpc.exceptions import (
+    RpcGenericServerDefinedError,
+    RpcInvalidRequestError,
+    RpcMethodNotFoundError,
+    RpcInvalidParamsError,
+    RpcInternalError,
+    RpcError,
+)
+# from aiohttp_json_rpc.communicaton import JsonRpcRequest
+
+from remme.shared.router import Router
+from remme.shared.exceptions import RemmeRpcError
+from remme.shared.messaging import Connection
+from .utils import load_methods
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class JsonRpc(JsonRpc):
+
+    def __init__(self, zmq_url, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._zmq_url = zmq_url
+        self._accepting = True
+        self._evthashes = {}
+        self._subsevt = {}
+
+        self._print_storage_state_task = weakref.ref(
```

Comment:
We should be able to disable that.

Reply:
I propose to left it, for monitoring objects to prevent a leaking.

Reply:
just add a feature toggle somewhere in configuration